src/wasm-exts/customModuleEncode.ml

```diff
@@ -841,16 +841,20 @@ let encode (em : extended_module) =
     let motoko_section_body labels =
       section 0 (vec string) labels (labels <> [])
 
-    let stable_types_body s =
-      string s
+    let utf8 bs =
+      try
+        let _ = Wasm.Utf8.decode bs in
+        put_string s bs
+      with Wasm.Utf8.Utf8 ->
+        assert false
```

Comment:
Nit: This assertion is somewhat overcomplicated. Why not
```suggestion
      ignore (Wasm.Utf8.decode bs);  (* assert well-formedness *)
      put_string s bs
```

Reply:
Don't I just get an uncaught exception then? Rather than a report of where the assertion failed. But ok.